app/api.py

```diff
@@ -75,17 +78,163 @@ def update(req: UserCreateRequest, token: UserToken) -> Empty:
+    is_me: bool = Field(title="リクエスト者と同じか")
+    is_host: bool = Field(title="ホストと同じか")
+
+
+class ResultUser(BaseModel):
+    user_id: int = Field(title="ユーザID")
+    judge_count_list: list[int] = Field(title="判定回数")
+    score: int = Field(title="スコア")
+
+
+class RoomInfoList(BaseModel):
+    room_info_list: list[RoomInfo] = Field(title="部屋情報の配列")
+
+
+class JoinRoomResult(BaseModel):
+    join_room_result: model.JoinRoomResult = Field(title="ルーム入場結果")
+
+
+class WaitRoomResult(BaseModel):
+    status: model.WaitRoomStatus = Field(title="結果")
+    room_user_list: list[RoomUser] = Field(title="ルームにいるユーザ一覧")
+
+
+class ResultRoomResult(BaseModel):
+    result_user_list: list[ResultUser] = Field(title="各ユーザの結果")
 
 
 class CreateRoomRequest(BaseModel):
     live_id: int
     select_difficulty: LiveDifficulty
 
 
+class ListRoomRequest(BaseModel):
+    live_id: int
+
+
+class JoinRoomRequest(BaseModel):
+    room_id: int
+    select_difficulty: LiveDifficulty
+
+
+class EmptyResult(BaseModel):
+    ...
+
+
+class WaitRoomRequest(BaseModel):
+    room_id: int
+
+
+class StartRoomRequest(BaseModel):
+    room_id: int
+
+
+class LeaveRoomRequest(BaseModel):
+    room_id: int
+
+
+class EndRoomRequest(BaseModel):
+    room_id: int
+    judge_count_list: list[int]
+    score: int
+
+
+class ResultRoomRequest(BaseModel):
+    room_id: int
+
+
+
 @app.post("/room/create")
 def create(token: UserToken, req: CreateRoomRequest) -> RoomID:
     """ルーム作成リクエスト"""
     print("/room/create", req)
     room_id = model.create_room(token, req.live_id, req.select_difficulty)
     return RoomID(room_id=room_id)
+
+
+@app.post("/room/list")
+def select(req: ListRoomRequest) -> RoomInfoList:
+    print("/room/list", req)
+    room_list = model.list_room(req.live_id)
+    if room_list is None:
+        return RoomInfoList(room_info_list=[])
+    room_list = list(map(lambda room: RoomInfo(
+            room_id=room.room_id,
+            live_id=room.live_id,
+            joined_user_count=room.joined_user_count,
+            max_user_count=room.max_user_count
+            ), room_list))
+    print(room_list)
+    return RoomInfoList(room_info_list=room_list)
+
+
+@app.post("/room/join")
+def join(token: UserToken, req: JoinRoomRequest) -> JoinRoomResult:
+    print("/room/join", req)
+    join_room_result = model.join_room(token, req.room_id, req.select_difficulty)
+    return JoinRoomResult(join_room_result=join_room_result)
+
+
+@app.post("/room/wait")
+def wait(token: UserToken, req: WaitRoomRequest) -> WaitRoomResult:
+    print("/room/wait", req)
+    status, members = model.wait_room(token, req.room_id)
+    members = list(map(lambda member: RoomUser(
+            user_id=member["user_id"],
+            name=member["name"],
+            leader_card_id=member["leader_card_id"],
+            select_difficulty=member["select_difficulty"],
+            is_me=member["is_me"],
+            is_host=member["is_host"]
+            ), members))
```

Comment:
リスト内包表記を使うと `list(map(lambda` よりも簡潔に書けます。

```suggestion
    members = [RoomUser(
            user_id=member["user_id"],
            name=member["name"],
            leader_card_id=member["leader_card_id"],
            select_difficulty=member["select_difficulty"],
            is_me=member["is_me"],
            is_host=member["is_host"]
            )
            for member in members]
```

あと、lambdaや内方で使う一時変数（ここでは member）は m とか1文字変数にしてしまった方が、
読みやすくなることが多いです。